packages/edit-post/src/store/selectors.js

```diff
@@ -562,36 +562,56 @@ export function areMetaBoxesInitialized( state ) {
  */
 export const getEditedPostTemplate = createRegistrySelector(
 	( select ) => () => {
-		const currentTemplate =
-			select( editorStore ).getEditedPostAttribute( 'template' );
+		const {
+			id: postId,
+			type: postType,
+			slug,
+			template: currentTemplate,
+		} = select( editorStore ).getCurrentPost();
+		const { getSite, getEditedEntityRecord, getEntityRecords } =
+			select( coreStore );
+		const siteSettings = getSite();
+		// First check if the current page is set as the posts page.
+		const isPostsPage = +postId === siteSettings?.page_for_posts;
+		if ( isPostsPage ) {
```

Comment:
This is the main change. Rest of the changes are due to the added object deconstruction..

Reply:
The +postId is to make sure postId is an integer, in which cases it is not an integer could we not safely assume it is always an integer like we are assuming page_for_posts is an integer?

Reply:
I'm not really sure what is the flow that could result in a string and I left that untouched for now. ). Probably some cases for comparing post entities ids and the site settings - I think I had come across something like that in the past.